maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/JarManifestForkConfiguration.java

```diff
@@ -150,10 +150,9 @@ private static String toClasspathElementUri( @Nonnull Path parent,
     {
         try
         {
-            return new URI( null,
-                            parent.relativize( classPathElement.toPath() ).toString().
-                                replace( File.separatorChar, '/' ),
-                            null ).toASCIIString();
+            Path relativePath = parent.relativize( classPathElement.toPath() );
```

Comment:
This won't tested by ITs because we cannot occupy any disk on Jenkins, but this algorithm could be tested by unit test and Mockito. At least Line 153 in mock and next two lines are real and tested. If the Line went to private static method, we could mock it via PowerMock, and then the next two lines together in another method calling the first static method. We would assert the returned value has Unix slashes and white spaces are escaped, paths are relative, etc.

Reply:
@jglick 
There is one IT test `Surefire855AllowFailsafeUseArtifactFileIT` which asserts Class-Path in manifest and surefire.properties, but it's hard to develope such test runing on any platform and any number of dists. Even in our case, using ASF Jenkins CI, we have workspace on `F:\` but the TEMP is on `C:\`.

Reply:
I cannot imagine you could write a reliable IT without Vagrant or similar.

Reply:
@jglick 
Not sure about Vagrant but I know that Docker works with Windows 10.
our Jenkins in ASF has Windows server 2008.